### ISSUE TITLE
[Model Enhancement]: American Sign Language Detection Model 

### ISSUE THREAD

**Issue by vinod-polinati** [abhisheks008/DL-Simplified#574]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : American Sign Language Detection </br>
:red_circle: **Aim** :  To Enhance the current  model using a good dataset and also use MediaPipe </br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/kapillondhe/american-sign-language </br>
:red_circle: **Approach** : I will enhance the model by using Tensorflow , MediaPipe , a larger dataset with more training data

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name :  Vinod Polinati
- GitHub Profile Link : https://github.com/vinod-polinati
- Email ID : [vinodd.polinati@gmail.com](vinodd.polinati@gmail.com)
- Participant ID (if applicable): Vinod Polinati (discord: nohoesplis)
- Approach for this Project :  I will enhance the model by using Tensorflow , MediaPipe , a larger dataset with more training data
- What is your participant role? Gssoc'24 Contributor

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by abhisheks008** [comment]
The existing project is having an accuracy score of 97%. Are you trying to enhance it more?

Here is the existing project, [American Sign Language Detection](https://github.com/abhisheks008/DL-Simplified/tree/main/American%20Sign%20Language%20Detection/models)

**Comment by vinod-polinati** [comment]
Yes I am trying to train the models more by using a much broader dataset.

**Comment by abhisheks008** [comment]
Then do one thing, build your model and add the same in the existing project folder. Will that be okay with you? As this is an enhancement task, you can't create a new project folder for this.

@vinod-polinati

**Comment by vinod-polinati** [comment]
Sure

**Comment by abhisheks008** [comment]
Yes you can go ahead with this issue. Assigned to you @vinod-polinati

**Comment by Adhivp** [comment]
Please @vinod-polinati notify after the work is finished, so that I can update the model in the web app I am currently building

**Comment by vinod-polinati** [comment]
Sure ! I'll make a pr as soon as possible

**Comment by abhisheks008** [comment]
> Sure ! I'll make a pr as soon as possible
> 
> @Adhivp

That's good to see you guys are coordinating with each other ❤️

**Comment by Adhivp** [comment]
The current model doesn't show much accuracy as labeled mobilenet 97%, when i Tested it couldn't even pass the images in training set , currently trying with VGG16 as per the guideliens , @abhisheks008 It would be nice if they added .h5 file along with the model notebooks. It takes a lot of time for web app development.

**Comment by abhisheks008** [comment]
> The current model doesn't show much accuracy as labeled mobilenet 97%, when i Tested it couldn't even pass the images in training set , currently trying with VGG16 as per the guideliens , @abhisheks008 It would be nice if they added .h5 file along with the model notebooks. It takes a lot of time for web app development.

Ahhh! I got it. No issues take your time, no need to hurry.

**Comment by Adhivp** [comment]
Updates?

**Comment by abhisheks008** [comment]
Hi @vinod-polinati please share your updates on the model enhancement for this project? As the web app creation issue is on hold for your enhancement, once you are done with the enhancement, the same model will be implemented in the web app deployment. Please try to understand and share your updates ASAP.

Otherwise this issue will be unassigned from you due to lack of activeness.

**Comment by abhisheks008** [comment]
Hi @vinod-polinati are you up for this issue? Your inactiveness caused a severe issue for the Web App deployment of the same project.

Please put some update on this.

**Comment by Adhivp** [comment]
Assign me

**Comment by khushi-igupta** [comment]
Hello @abhisheks008  I am a GsSOC'24 contributor and I also would like to work on this issue. Can you please assign it to me?

**Comment by abhisheks008** [comment]
> Assign me

Assigned to you @Adhivp

**Comment by abhisheks008** [comment]
> Hello @abhisheks008 I am a GsSOC'24 contributor and I also would like to work on this issue. Can you please assign it to me?

Hi @khushi-igupta this issue is already assigned to a contributor. You can find out other open issues present here in the Issues section.

**Comment by Adhivp** [comment]
Hey should I focus only on  numbers also? (or only in alphabets of American sign language), and also can I mix 2/3 datasets together? @abhisheks008
